FunctionalArray.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FunctionalArray.swift
+//  SwiftTrainingExercise
+//
+//  Created by Ahmed Ibrahim on 20/06/2023.
+//
+
+import Foundation
+
+class FunctionalArray<Element> where Element: Equatable & Comparable & Hashable {
+  
+  var elementsArray: [Element] = [] // how to set this as private only for subclass and superclass so its not accessable ?
+  private(set) var maxSize = Int.max;
+  
+  init() {}
+  
+  init(maxSize size: Int) throws {
+    guard size >= 0 else {
+      throw DataStructureError.negativeSize
+    }
+    self.maxSize = size
+  }
+  
+  func sort(by sortFunction: (Element, Element) -> Bool) {
+    elementsArray.sort(by: sortFunction)
+  }
+  
+  func removeDuplicates(){
+    var set = Set<Element>()
+    var uniqueElementsArray: [Element] = []
+    for element in elementsArray {
+      if set.insert(element).inserted {
+        uniqueElementsArray.append(element)
+      }
+    }
+    self.elementsArray = uniqueElementsArray
```

Comment:
how do you preserve order?

Reply:
elements  are added from the start of the original array till the end, if one is found be duplicate the append will be skipped, so for duplicate elements the first one that is added is the one that will be kept.